module/zfs/zfs_chksum.c

```diff
@@ -195,19 +226,36 @@ chksum_benchit(chksum_stat_t *cs)
 	chksum_run(cs, abd, ctx, 3, &cs->bs16k);
 	chksum_run(cs, abd, ctx, 4, &cs->bs64k);
 	chksum_run(cs, abd, ctx, 5, &cs->bs256k);
+
+	/* check if we ran on a slow cpu */
+	if (limit == LIMIT_INIT) {
+		if (cs->bs1k < 300) {
+			limit = LIMIT_NEEDED;
+			chksum_kstat_headers = chksum_kstat_headers_mini;
+		} else {
+			limit = LIMIT_NOLIMIT;
+			chksum_kstat_headers = chksum_kstat_headers_full;
+		}
+	}
+
+	/* skip benchmarks >= 1MiB, when edonr is slower then 300 MiB */
```

Comment:
What makes this specific to edonr?  It seems like this is only mostly true since edonr gets benchmarked first.  If that were to change then this would no longer be accurate.

Reply:
Yes, that is correct, I will add a comment that edonr needs to be the first one.